File: app/src/main/java/com/hnidesu/taskmanager/ui/EditTextEx.java
package com.hnidesu.taskmanager.ui;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatEditText;

public class EditTextEx extends AppCompatEditText {
    public interface TextChangeListener{
        void onTextChange(View view, CharSequence text);
    }
    private TextChangeListener mTextChangeListener;
    public EditTextEx(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public void setTextChangeListener(TextChangeListener textChangeListener) {
        this.mTextChangeListener = textChangeListener;
    }

    @Override
    protected void onTextChanged(CharSequence text, int start, int lengthBefore, int lengthAfter) {
        super.onTextChanged(text, start, lengthBefore, lengthAfter);
        if(mTextChangeListener!=null)
            mTextChangeListener.onTextChange(this,text);
    }
}

File: app/src/main/java/com/hnidesu/taskmanager/widget/view/EditTextEx.kt
package com.hnidesu.taskmanager.widget.view

import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.appcompat.widget.AppCompatEditText

class EditTextEx(context: Context, attrs: AttributeSet?) : AppCompatEditText(
    context, attrs
) {
    private var mTextChangeListener: TextChangeListener? = null

    interface TextChangeListener {
        fun onTextChange(view: View?, text: CharSequence?)
    }

    fun setTextChangeListener(textChangeListener: TextChangeListener?) {
        mTextChangeListener = textChangeListener
    }

    override fun onTextChanged(
        text: CharSequence,
        start: Int,
        lengthBefore: Int,
        lengthAfter: Int
    ) {
        super.onTextChanged(text, start, lengthBefore, lengthAfter)
        mTextChangeListener?.onTextChange(this, text)
    }
}
